### DISCUSSION TITLE
React Setup

### DISCUSSION THREAD

**Discussion by NicholasGannon** [community/community#136890] [Programming Help]
### Select Topic Area

Question

### Body

When starting a **new React project** using Vite: Are there any unwanted files, images, etc... that can be removed + aren't necessary so that I can start with a clean slate?

**Comment by ebndev** [comment]
Hi @NicholasGannon, thanks for posting in GitHub Discussions! 

I've gone ahead and moved this post to our `Programming Help` category, as this topic may be more relevant there and might help someone give you a nudge in the right direction. 😁

**Comment by itdev-1210** [comment]
Hi, @NicholasGannon 

When starting a new React project using Vite, you can remove the following files and content to start with a clean slate:

**1. Unnecessary Assets:**

src/assets folder: This often contains a Vite logo or other default images. You can remove the entire folder if you don't need it.
src/favicon.svg: The default Vite favicon can be replaced or removed if you're not using it.

**2. Boilerplate Code:**

src/App.css: Contains default styling for the sample components. You can remove or clear the file if you plan to write your own styles.
src/App.tsx (or App.jsx): Remove the boilerplate content and replace it with your own component structure.
src/main.tsx (or main.jsx): You might want to clear out any default imports or code that relates to the Vite-specific setup.

**3. Test Files:**

src/tests folder (if it exists): Sometimes projects might include test examples that you may not need. Remove any files here if you're not planning to use them.

**4. CSS Frameworks:**

index.css: If you are using a different styling method (like CSS-in-JS, TailwindCSS, etc.), you might not need this file.

**5. Readme:**

README.md: Update the readme to reflect your own project information.

**6. Other Configuration Files:**

tsconfig.json (if using TypeScript): Modify or remove any default paths or settings that aren't needed for your project.

After cleaning up these files, you'll have a much leaner project structure that's ready for your custom development.

**Comment by NicholasGannon** [comment]
Hi @itdev-1210 
Thank you so much for the response! This is definitely very helpful!

**Reply by itdev-1210** [reply]
You are welcome.

**Comment by mtm-zakir** [comment]
When starting a new React project using Vite, there are a few files and assets that can be safely removed to start with a clean slate. Here's a list of files and folders you can remove or clean up:

1. **Unused Assets:**
   - Inside the `src/assets/` folder, you might find images or other assets (e.g., `vite.svg`) that are not required for your project. You can safely delete these unless you plan to use them.

2. **Styles:**
   - If there's a `src/App.css` or similar CSS file that's included by default but you're not using it, you can remove or replace it with your own styling approach (e.g., CSS, SCSS, or styled-components).

3. **App Component Boilerplate:**
   - The default content inside `src/App.jsx` (or `.tsx` if you're using TypeScript) often contains sample JSX like a logo and a paragraph. You can clear this out and start fresh.

4. **Public Directory:**
   - The `public/` folder may contain files like `favicon.svg` or other default assets. If you have your own assets, you can replace or delete these.

5. **ReadMe Files:**
   - Any auto-generated README files in the root folder that do not match your project's documentation needs can be deleted or customized.

6. **Test Files (if any):**
   - If you don’t plan on writing tests immediately, you can delete any test files (like `src/App.test.jsx`), although it's good practice to write tests eventually.

After removing these files, your project will be cleaner and ready for customization.

**Reply by NicholasGannon** [reply]
Hi @mtm-zakir 
Thank you so much for the help - Much appreciated!

**Comment by HiepF5** [comment]
Unused Assets:

Inside the src/assets/ folder, you might find images or other assets (e.g., vite.svg) that are not required for your project. You can safely delete these unless you plan to use them.
Styles:

If there's a src/App.css or similar CSS file that's included by default but you're not using it, you can remove or replace it with your own styling approach (e.g., CSS, SCSS, or styled-components).
App Component Boilerplate:

The default content inside src/App.jsx (or .tsx if you're using TypeScript) often contains sample JSX like a logo and a paragraph. You can clear this out and start fresh.
Public Directory:

The public/ folder may contain files like favicon.svg or other default assets. If you have your own assets, you can replace or delete these.
ReadMe Files:

Any auto-generated README files in the root folder that do not match your project's documentation needs can be deleted or customized.
Test Files (if any):

If you don’t plan on writing tests immediately, you can delete any test files (like src/App.test.jsx), although it's good practice to write tests eventually.

**Reply by NicholasGannon** [reply]
Hi @HiepF5 
Thanks for the response!

**Comment by QuentinWatt** [comment]
I think in general, all the files in the Vite Setup are there to show you where you should place your own replacements. They're all "useful" in that way.

You would replace `src/App.css` with your own css.
You would replace `src/assets` with your own website logo and other assets.

You will of course be replacing the boilerplace code in `src`, and add your own files to `src`. Possibly also organise all components and custom hooks into folders within that directory. 

So I don't think anything that comes shipped in the Vite Template is unnecessary waste, just the minimum required for a guideline.

**Reply by NicholasGannon** [reply]
Thank you! This was very helpful!

**Comment by igorpereirag** [comment]
When you create a new React project using Vite, there are a few files and assets that can be removed if you're looking to start with a clean slate. These files typically come from the default template and aren't necessary for the project’s core functionality. Here’s a list of what you can safely remove:

1. Logo and Vite Assets
  Located in: src/assets/
  Files:
  vite.svg or any other images (if included).
  These are just placeholders or logos included in the default template.
2. Default Styles
  Located in: src/
  Files:
  App.css
  index.css (if it exists)
These are basic styles for the default template. If you’re starting fresh with custom styles or a CSS framework, they can be removed.

**Reply by NicholasGannon** [reply]
Thank you for the response! Much appreciated!

**Comment by lqduy** [comment]
When starting a new React project with Vite, you can clean up the following files and directories that aren't essential for a minimal setup:

1. **`src/assets`**: This folder typically contains the default Vite logo or other images that you don't need. You can remove the folder or any images inside it.

2. **`src/App.css`**: If you're not using the default CSS provided by Vite, you can remove or replace this file.

3. **`src/App.tsx` or `App.jsx`**: The default app content often has boilerplate code and text. You can clear this out to write your own logic.

4. **`src/index.css`**: If you prefer to manage styles differently or with another library like Tailwind CSS or styled-components, you can remove this file.

5. **`public` folder**: Sometimes, the public folder contains unnecessary assets (like a `favicon` or logos). If you're not using these, you can clean them up.

6. **`README.md`**: Depending on your workflow, you might want to clear out the initial content if it's not relevant.

After this cleanup, you’ll have a minimal project ready to start building!

**Reply by NicholasGannon** [reply]
Thank you - Much appreciated!
